test/unit/Governor.js

```diff
@@ -0,0 +1,558 @@
+
+            await governor.stage(
+                {
+                    target: [setUint256.address],
+                    value: ["0"],
+                    data: [data]
+                },
+                "100"
+            )
+
+            await fixture.rpc.wait(100)
+
+            const tx = await governor.execute(
+                {
+                    target: [setUint256.address],
+                    value: ["0"],
+                    data: [data]
+                }
+            )
+
+            truffleAssert.eventEmitted(
+                tx,
+                "UpdateExecuted",
+                e => e.update.target[0] == setUint256.address
+                    && e.update.value[0] == "0"
+                    && e.update.data[0] == data,
+                "UpdateStaged event not emitted correctly"
+            )
+        })
+    })
+
+    describe("batch executeUpdate", () => {
+        it("reverts when the update has not been staged", async () => {
+            const {data: data0} = await setUint256Tx("0", accounts[0])
+            const {data: data1} = await setUint256Tx("1", accounts[0])
+            const {data: data2} = await setUint256Tx("5", accounts[0])
+
+            // stage the update partially
+            await governor.stage(
+                {
+                    target: [setUint256.address],
+                    value: ["0"],
+                    data: [data0]
+                },
+                "100"
+            )
+
+            await truffleAssert.reverts(
+                governor.execute(
+                    {
+                        target: [setUint256.address, setUint256.address, setUint256.address],
+                        value: ["0", "0", "0"],
+                        data: [data0, data1, data2]
+                    }
+                ),
+                "update is not staged"
+            )
+        })
+
+        it("reverts when delay for the staged update has not expired", async () => {
+            const {data: data0} = await setUint256Tx("0", accounts[0])
+            const {data: data1} = await setUint256Tx("1", accounts[0])
+            const {data: data2} = await setUint256Tx("5", accounts[0])
+
+            await governor.stage(
+                {
+                    target: [setUint256.address, setUint256.address, setUint256.address],
+                    value: ["0", "0", "0"],
+                    data: [data0, data1, data2]
+                },
+                "100"
+            )
+
+            await truffleAssert.reverts(
+                governor.execute(
+                    {
+                        target: [setUint256.address, setUint256.address, setUint256.address],
+                        value: ["0", "0", "0"],
+                        data: [data0, data1, data2]
+                    }
+                ),
+                "delay for update not expired"
+            )
+        })
+
+        it("reverts when one of the remote calls in the batch fails", async () => {
```

Comment:
Should we add a test when a remote call other than the first one reverts that checks that the revert reason corresponds with the correct remote call (and not just the first one)?

Reply:
I don't see much value here because 

1. execute already tests that `success == true` when the call doesn't fail and thus no revert is encountered
2. we will only ever get the revert reason for the first failing call

Reply:
Fair enough. After some more reflection, I think I'm actually more interested in seeing cases where different the transactions in an update are for different contract and/or different functions - maybe something simple can be included in an integration test.

Reply:
+1 for some integration test scenario's using our actual contracts

Reply:
I added integration tests but I also added some changes for Testability as mentioned in this comment https://github.com/livepeer/protocol/pull/391#issuecomment-674029279 , I also removed the `whenSystemNotPaused` modifier from `Minter.DepositEth` like in #391. 

These Minter changes for testability are a separate commit and can be dropped if we merge #391 first (and the  revert reasons mentioned are added).

Reply:
I also decided to not  test for all `Minter.migrateToNewMinter`  revert paths since that is not the purpose of this test, I added two cases here and I think that's sufficient.